partiql-tests-data/eval/primitives/operators/overlaps.ion

```diff
@@ -0,0 +1,779 @@
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with long periods",
+    statement:"[DATE '2020-01-01', DATE '2025-12-31'] OVERLAPS [DATE '2023-01-01', DATE '2030-12-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with overlapping periods ROW value",
+    statement:"ROW(DATE '2023-01-01', DATE '2023-01-31') OVERLAPS ROW(DATE '2023-01-15', DATE '2023-02-15')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with non-overlapping periods ROW value",
+    statement:"ROW(DATE '2023-01-01', DATE '2023-01-31') OVERLAPS ROW(DATE '2023-02-01', DATE '2023-02-28')",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with mixed ROW and array constructors",
+    statement:"(DATE '2023-01-01', DATE '2023-01-31') OVERLAPS [DATE '2023-01-15', DATE '2023-02-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with null as first argument",
+    statement:"null OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with null as first argument",
+    statement:"null OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with null as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS null",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with null as second argument",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS null",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with missing as first argument",
+    statement:"MISSING OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with missing as first argument",
+    statement:"MISSING OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
```

Comment:
I think the correct behavior is to return `MISSING` in both eval modes rather than return `null` in coerce mode and `missing` in error mode.

PartiQL will propagate MISSING for operations and functions that having `MISSING` as one of the argument values.

```
partiql ▶ 1 + MISSING;
w: [semantic] Expression always returns missing

=== RESULT ===
missing

OK!
partiql ▶ MISSING IN [1, 2, 3];
w: [semantic] Expression always returns missing

=== RESULT ===
missing

OK!
```

Reply:
Let's chat about this more. Based on the spec definition of rewriting to a sequence of logical operators, I think `missing` would act like `null` here and actually return `true`.